cluster/ci/origin/logging-infra/finished.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"time"
+
+	"os"
+	"path/filepath"
+
+	"cloud.google.com/go/storage"
+)
+
+type Finished struct {
+	// Timestamp is the time at which we finish
+	// execution, in seconds past the UNIX epoch
+	Timestamp int64 `json:"timestamp"`
+
+	// Passed is the result of the build
+	Passed bool `json:"passed"`
+}
+
+func UploadFinishedData(processLog, metadataFile, artifactDir string, passed bool, gcsBucket *storage.BucketHandle) error {
+	metadata, err := loadMetadata(metadataFile)
+	if err != nil {
+		return err
+	}
+
+	finishedFile, err := writeFinishedMetadata(passed)
+	if err != nil {
+		return err
+	}
+
+	gcsPath := metadata.GcsPath()
+	uploadTargets := map[string]string{
+		finishedFile: gcsPath + "/finished.json",
+		// TODO(skuznets): we want to stream this log during the run
```

Comment:
I'd imagine the stream would be exposed in Gubernator or any other tools people build to interface with GCS/job artifacts+logging storage in general. Unless the vision is that people go to Deck to view jobs+artifacts+failures, it seems like we want to have the interface at the GCS layer